config/config.go

```diff
@@ -163,19 +168,28 @@ func (l *Loader) Paths() []string {
 }
 
 // SetConfigFiles overrides the set of available config files for the service.
+// Configs will be interpolated with values from environment variables.
 func (l *Loader) SetConfigFiles(files ...string) {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
 	l.configFiles = files
 }
 
+// SetStaticConfigFiles overrides the set of available config files for the service.
+// Config values will not be interpolated.
+func (l *Loader) SetStaticConfigFiles(files ...string) {
+	l.lock.Lock()
```

Comment:
if you embed sync.RWMutex, you can just call Lock on loader directly. So l.Lock() instead of l.lock.Lock()